src/typing/type_inference_js.ml

```diff
@@ -4674,6 +4680,7 @@ and mk_body id cx param_types_map param_types_loc ret body this super =
           (create_env_entry ret ret None)
   )
   in
+  let block = { scope = Hoist; entries } in
```

Comment:
The spec states that class bodies have a normal "lexical environment" (aka this should probably be `BlockScope` rather than `HoistedScope`)

https://people.mozilla.org/~jorendorff/es6-draft.html#sec-runtime-semantics-classdefinitionevaluation